app/policies/suite_policy.rb

```diff
@@ -0,0 +1,28 @@
+class SuitePolicy < ApplicationPolicy
+    class Scope < Scope
+      def resolve
+        scope.all
+      end
+    end
+  
+    def new?
+      user.is_admin
+    end
+  
+    def show?
+      true
+    end
+  
+    def create?
+      user.is_admin
+    end
+  
+    def update?
+      user == record || user.is_admin
```

Comment:
This does not make sense. Should be user.is_admin only. user == record means "if the current user is the dorm he is trying to modify".